src/pages/Notifications/NotificationsLog/Page.tsx

```diff
@@ -0,0 +1,198 @@
+
+import { NotificationsLogDateFilterValue } from '../../../components/Notifications/NotificationsLog/NotificationsLogDateFilter';
+import NotificationsLogTable, {
+  DrawerType,
+  PaginationType,
+} from '../../../components/Notifications/NotificationsLog/NotificationsLogTable';
+import NotificationsLogToolbar from '../../../components/Notifications/NotificationsLog/NotificationsLogToolbar';
+import { PageHeader } from '../../../components/PageHeader';
+import { Messages } from '../../../properties/Messages';
+
+export type NotificationsPeriod = [Date | undefined, Date | undefined];
+
+const sortMapper = {
+  0: 'title',
+  2: 'time',
+};
+
+const RETENTION_DAYS = 14;
+
+const createFilter = (dateFilter, period) => {
+  let filterPeriod = [undefined, undefined] as [
+    Date | undefined,
+    Date | undefined
+  ];
+  const today = toUtc(new Date());
+  const DATE_FORMAT = 'yyyy-MM-dd';
+  const filter = new Filter();
+
+  let yesterday;
+  switch (dateFilter) {
+    case NotificationsLogDateFilterValue.LAST_14:
+      filterPeriod = [
+        sub(toDate(today), {
+          days: 14,
+        }),
+        today,
+      ];
+      break;
+    case NotificationsLogDateFilterValue.LAST_7:
+      filterPeriod = [
+        sub(toDate(today), {
+          days: 7,
+        }),
+        today,
+      ];
+      break;
+    case NotificationsLogDateFilterValue.TODAY:
+      filterPeriod = [today, today];
+      break;
+    case NotificationsLogDateFilterValue.YESTERDAY:
+      yesterday = sub(toDate(today), {
+        days: 1,
+      });
+      filterPeriod = [yesterday, yesterday];
+      break;
+    case NotificationsLogDateFilterValue.CUSTOM:
+      filterPeriod = period;
+      break;
+  }
+
+  if (filterPeriod[0] && filterPeriod[1]) {
+    filter.and('start', Operator.EQUAL, format(filterPeriod[0], DATE_FORMAT));
+    filter.and('end', Operator.EQUAL, format(filterPeriod[1], DATE_FORMAT));
+  }
+
+  return {
+    startDate: filterPeriod[0]
+      ? format(filterPeriod[0], DATE_FORMAT)
+      : undefined,
+    endDate: filterPeriod[1] ? format(filterPeriod[1], DATE_FORMAT) : undefined,
+  };
+};
+
+export const NotificationsLogPage: React.FunctionComponent = () => {
+  const [data, setData] = useState<DrawerType>([]);
+  const [period, setPeriod] = useState<NotificationsPeriod>([
+    undefined,
+    undefined,
+  ]);
+  const [dateFilter, setDateFilter] = useState<NotificationsLogDateFilterValue>(
+    NotificationsLogDateFilterValue.LAST_14
+  );
+  const [pagination, setPagination] = useState<PaginationType>({
+    offset: 0,
+    limit: 20,
+    count: 0,
+  });
+  const [sort, setSort] = useState<{
+    columnIndex: number;
+    direction: 'asc' | 'desc';
+  }>({
+    columnIndex: 2,
+    direction: 'desc',
+  });
+
+  const filterBuilder = useMemo(
+    () => createFilter(dateFilter, period),
+    [dateFilter, period[0]?.toString(), period[1]?.toString()]
+  );
+
+  const callApi = useCallback((pagination, sort, filterBuilder) => {
```

Comment:
Please add `TODO: use new JS client instead of directly calling fetch` above this callback.

Reply:
fixed